src/plugins/offline/azureOfflinePlugin.ts

```diff
@@ -0,0 +1,48 @@
+import AzureProvider from "../../provider/azureProvider";
+import Serverless from "serverless";
+import { ConverterService } from "../../services/converterService"
+import { OfflineService } from "../../services/offlineService";
+
+
+export class AzureOfflinePlugin {
+  public hooks: { [eventName: string]: Promise<any> };
+  public commands: any;
+  private offlineService: OfflineService;
+
+  public constructor(private serverless: Serverless, private options: Serverless.Options) {
+    this.hooks = {
+      "before:offline:offline": this.azureOfflineBuild.bind(this),
```

Comment:
Running build before `offline` and as it's own command `offline build`

Reply:
It's different - seems to be much more verbose than what we would need. My vision here is (and please let me know if you have a different one):

```
sls offline build     (run the conversion put the proper metadata in the respective function.json files)
sls offline      (run the build and then spawn the `npm start` process)
```

Reply:
Maybe the offline plugin just does the build and the docs just say to run `npm start`. Open to suggestions here